packages/m2m-gateway/src/services/delegationService.ts

```diff
@@ -1,6 +1,54 @@
+import { delegationApi, m2mGatewayApi } from "pagopa-interop-api-clients";
+import { WithLogger } from "pagopa-interop-commons";
 import { PagoPAInteropBeClients } from "../clients/clientsProvider.js";
+import { pollResource } from "../utils/polling.js";
+import { M2MGatewayAppContext } from "../utils/context.js";
+import { assertMetadataExists } from "../utils/validators/validators.js";
+import { toM2MGatewayApiConsumerDelegation } from "../model/delegationApiConverter.js";
+import { assertDelegationKindIs } from "../utils/validators/delegationValidators.js";
 
 // eslint-disable-next-line @typescript-eslint/explicit-function-return-type
-export function delegationServiceBuilder(_clients: PagoPAInteropBeClients) {
-  return {};
+export function delegationServiceBuilder({
+  delegationProcessClient,
+}: PagoPAInteropBeClients) {
+  // eslint-disable-next-line @typescript-eslint/explicit-function-return-type
+  const pollDelegation = (
+    delegationId: delegationApi.Delegation["id"],
+    headers: M2MGatewayAppContext["headers"]
+  ) =>
+    pollResource(() =>
+      delegationProcessClient.delegation.getDelegation({
+        params: { delegationId },
+        headers,
+      })
+    );
+
+  return {
+    async createConsumerDelegation(
+      seed: m2mGatewayApi.DelegationSeed,
+      { headers }: WithLogger<M2MGatewayAppContext>
+    ): Promise<m2mGatewayApi.ConsumerDelegation> {
+      const response =
+        await delegationProcessClient.consumer.createConsumerDelegation(seed, {
+          headers,
+        });
+
+      const polledResource = await pollDelegation(
+        response.data.id,
+        headers
+      )({
+        // TODO maybe this is general enough and could be moved to the polling function?
+        checkFn: (polled) => {
+          assertMetadataExists(response);
```

Comment:
can we remove this check, considering it is already done [here](https://github.com/pagopa/interop-be-monorepo/pull/1777/files?diff=split&w=1#diff-9624b7d3b619b796b0cf61d269ad56e6d46d4d6e501d7ad595b9b814d2fcac55R35)?

Reply:
It's actually different, the one you linked is on the polled resource (the result of the GET), while this one is on the response of the POST creation request.
Do you think it's worth already hardcoding this check function as a standard in the polling? Having this `polled.metadata.version >= response.metadata.version` check become the "default" option somehow?

Reply:
my bad, looks good as it is